src/shims/unix/linux/eventfd.rs

```diff
@@ -28,16 +30,33 @@ struct Event {
     counter: u64,
     is_nonblock: bool,
     clock: VClock,
+    /// We have to store our own FdID in contrast to every other file descriptor out there, because
+    /// we are updating ourselves when writing and reading. Technically `Event` is like socketpair, but
+    /// it does not create two separate file descriptors. Thus we can't re-borrow ourselves via
+    /// `FileDescriptionRef::check_and_update_readiness` while already being mutably borrowed for read/write.
+    id: FdId,
```

Comment:
This is unfortunate...

I think I would prefer passing the `FdId` to `fn read` and `fn write`. That avoids having redundant long-term state which has a risk of getting out-of-sync. (EDIT: or pass the `FileDescriptorRef`, also see my other comment below)

There's probably a nicer way to achieve this but I can't think of one right now.^^

Reply:
I made an attempt to allow implementing `FileDescription` for `WithFdId<EventFd>`, but it can't be done nicely. Passing FdId on each call is a bit ugly, but not the end of the world.